qa/tasks/ceph.py

```diff
@@ -610,195 +600,258 @@ def cluster(ctx, config):
         Mkfs osd nodes.
         Add keyring information to monmaps
         Mkfs mon nodes.
-
     On exit:
         If errors occurred, extract a failure message and store in ctx.summary.
         Unmount all test files and temporary journaling files.
         Save the monitor information and archive all ceph logs.
         Cleanup the keyring setup, and remove all monitor map and data files left over.
-
     :param ctx: Context
     :param config: Configuration
     """
-    if ctx.config.get('use_existing_cluster', False) is True:
-        log.info("'use_existing_cluster' is true; skipping cluster creation")
-        yield
 
     testdir = teuthology.get_testdir(ctx)
     cluster_name = config['cluster']
     data_dir = '{tdir}/{cluster}.data'.format(tdir=testdir, cluster=cluster_name)
-    log.info('Creating ceph cluster %s...', cluster_name)
-    log.info('config %s', config)
-    log.info('ctx.config %s', ctx.config)
-    run.wait(
-        ctx.cluster.run(
-            args=[
-                'install', '-d', '-m0755', '--',
-                data_dir,
-            ],
-            wait=False,
-        )
-    )
+    default_conf_path = '/etc/ceph/{cluster}.conf'.format(cluster=cluster_name)
+    conf_path = config.get('conf_path', default_conf_path)
+    default_keyring = '/etc/ceph/{cluster}.keyring'.format(cluster=cluster_name)
+    keyring_path = config.get('keyring_path', default_keyring)
+    firstmon = teuthology.get_first_mon(ctx, config, cluster_name)
+    monmap_path = '{tdir}/{cluster}.monmap'.format(tdir=testdir,
+                                                   cluster=cluster_name)
+    osds = ctx.cluster.only(teuthology.is_type('osd', cluster_name))
 
-    run.wait(
-        ctx.cluster.run(
+    if not hasattr(ctx, 'ceph'):
+        ctx.ceph = {}
+    ctx.ceph[cluster_name] = argparse.Namespace()
+
+    backup_ext = '.orig.teuthology'
+    devs_to_clean = {}
+    conf = configobj.ConfigObj()
+
+    if ctx.config.get('use_existing_cluster', False) is True:
+        log.info("'use_existing_cluster' is true; skipping cluster creation")
+
+        log.info("Backup ceph configuration file...")
+        remote = ctx.cluster.remotes.keys()[0]
```

Comment:
Why do we do this only on first remote? My guess is `use_existing_cluster` is constrained to the single machine scenario only but, if so, it might be good idea to enforce such an assumption with e.g.

```python
  assert len(remotes) == 1
```

Reply:
I think the ceph.conf of all nodes is the same, so only the first remote is backed up. After the test, the original ceph.conf will be copied to other nodes.